8.tcga-classify/classify-cancer-types.py

```diff
@@ -0,0 +1,234 @@
+signals = ["signal", "shuffled"]
+alphas = [0.1, 0.13, 0.15, 0.2, 0.25, 0.3]
+l1_ratios = [0.15, 0.16, 0.2, 0.25, 0.3, 0.4]
+
+# Load data to build y matrices
+base_url = "https://github.com/greenelab/pancancer/raw"
+commit = "2a0683b68017fb226f4053e63415e4356191734f"
+
+# Load data
+file = "{}/{}/data/sample_freeze.tsv".format(base_url, commit)
+sample_freeze_df = pd.read_table(file, index_col=0)
+
+file = "{}/{}/data/mutation_burden_freeze.tsv".format(base_url, commit)
+mut_burden_df = pd.read_table(file, index_col=0)
+
+# Track total metrics for each gene in one file
+full_metrics_list = []
+count_list = []
+
+# Obtain a dictionary of file directories for loading each feature matrix (X)
+z_matrix_dict = build_feature_dictionary()
+
+# Provide one vs all classifications for all 33 different cancertypes in TCGA
+for acronym in sample_freeze_df.DISEASE.unique():
+
+    # Create list to store cancer-type specific results
+    cancertype_auc_list = []
+    cancertype_aupr_list = []
+    cancertype_coef_list = []
+
+    # Create directory for the cancer-type
+    cancertype_dir = os.path.join("results", "cancer-type", acronym)
+    os.makedirs(cancertype_dir, exist_ok=True)
+
+    y_df, count_df = process_y_matrix_cancertype(
+        acronym=acronym,
+        sample_freeze=sample_freeze_df,
+        mutation_burden=mut_burden_df,
+        hyper_filter=5,
+    )
+
+    # Track the status counts of all classifiers
+    count_list.append(count_df)
+
+    # Now, perform all the analyses for each X matrix
+    for signal in z_matrix_dict.keys():
+        z_dim_dict = z_matrix_dict[signal]
+        for z_dim in z_dim_dict.keys():
+            seed_z_dim_dict = z_dim_dict[z_dim]
+            for seed in seed_z_dim_dict.keys():
+                z_train_file = z_matrix_dict[signal][z_dim][seed]["train"]
+                z_test_file = z_matrix_dict[signal][z_dim][seed]["test"]
+
+                for alg in algorithms:
+                    # Load and process data
+                    train_samples, x_train_df, y_train_df = align_matrices(
+                        x_file=z_train_file,
+                        y=y_df,
+                        add_cancertype_covariate=False,
+                        algorithm=alg,
+                    )
+
+                    test_samples, x_test_df, y_test_df = align_matrices(
+                        x_file=z_test_file,
+                        y=y_df,
+                        add_cancertype_covariate=False,
+                        algorithm=alg,
+                    )
+
+                    # Train the model
+                    print(
+                        "Training model... cancer-type: {}, "
+                        "algorithm: {}, signal: {}, z_dim: {}, "
+                        "seed: {}".format(acronym, alg, signal, z_dim, seed)
+                    )
+
+                    # Fit the model
+                    cv_pipeline, y_pred_train_df, y_pred_test_df, y_cv_df = train_model(
```

Comment:
How robust are your hyperparam fits using CV?

Reply:
They are fairly variable. And their variability varies a lot depending on the classification task. This analysis takes a while to run, so I sweep over only a few combinations. Ideally I would sweep over more.